src/Aspire.Hosting.Azure.KeyVault/AzureKeyVaultResourceExtensions.cs

```diff
@@ -32,23 +34,52 @@ public static IResourceBuilder<AzureKeyVaultResource> AddAzureKeyVault(this IDis
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <param name="configureResource">Callback to configure the underlying <see cref="global::Azure.Provisioning.KeyVaults.KeyVault"/> resource.</param>
+    /// <param name="configureResource">Callback to configure the underlying <see cref="global::Azure.Provisioning.KeyVault.KeyVaultService"/> resource.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
     [Experimental("AZPROVISION001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
-    public static IResourceBuilder<AzureKeyVaultResource> AddAzureKeyVault(this IDistributedApplicationBuilder builder, string name, Action<IResourceBuilder<AzureKeyVaultResource>, ResourceModuleConstruct, KeyVault>? configureResource)
+    public static IResourceBuilder<AzureKeyVaultResource> AddAzureKeyVault(this IDistributedApplicationBuilder builder, string name, Action<IResourceBuilder<AzureKeyVaultResource>, ResourceModuleConstruct, KeyVaultService>? configureResource)
     {
         builder.AddAzureProvisioning();
 
         var configureConstruct = (ResourceModuleConstruct construct) =>
         {
-            var keyVault = construct.AddKeyVault(name: construct.Resource.Name);
-            keyVault.AddOutput("vaultUri", x => x.Properties.VaultUri);
+            var keyVault = new KeyVaultService(construct.Resource.Name)
+            {
+                Location = construct.AddLocationParameter(),
+                Properties = new KeyVaultProperties()
+                {
+                    TenantId = new MemberExpression(new FunctionCallExpression(new IdentifierExpression("tenant")), "tenantId"),
```

Comment:
Shouldn't this be a built-in bicep function?

@tg-msft What does this turn into?

Reply:
Yes, this turns into `tenant().tenantid`.  We just haven't added it to `BicepFunction` yet.

Reply:
Fixed.